packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-gateway.ts

```diff
@@ -54,6 +55,13 @@ export default class GooglePayPaypalCommerceGateway extends GooglePayGateway {
         await this.paypalCommerceScriptLoader.getPayPalSDK(paymentMethod, currency);
 
         this.googlepayConfig = await this.paypalCommerceScriptLoader.getGooglePayConfigOrThrow();
+
+        const { allowedPaymentMethods } = this.googlepayConfig;
+
+        this.setGatewayIdentifier(
+            allowedPaymentMethods[0]?.tokenizationSpecification?.parameters?.gateway ||
+                this.defaultGateway,
```

Comment:
We can avoid applying `this.defaultGateway` variable here if you make some update for `setGatewayIdentifier` method:

```
protected setGatewayIdentifier(gateway?: string) {
    this._gatewayIdentifier = gateway || this._gatewayIdentifier;
}
```

What do you think?

Reply:
@bc-nick Thanks for comment! But I would like this method to keep a simple `setter` logic. Especially it's a parent class.
I think we can keep logic with default variables outside of this method.

Reply:
@bc-dronov you already set this variable here https://github.com/bigcommerce/checkout-sdk-js/blob/de326bd05ee167b5636ee87e78d96abd1312ee86/packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-gateway.ts#L29

so you do not need to pass it to `setGatewayIdentifier` because parent class already has it

Reply:
@serhii-tkachenko Agree, we should don't forget to keep it simple.

Reply:
@bc-nick I think we don't need to change parent's method logic because of child's class. If it would be possible, I would call the `super` method like:
`super(this.defaultGateway, service)`
But unfortunately `super` must be called before accessing  `this`.

Reply:
To make it more easily there is another way. @bc-dronov you can do the same logic here https://github.com/bigcommerce/checkout-sdk-js/blob/de326bd05ee167b5636ee87e78d96abd1312ee86/packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-gateway.ts#L74

For example: 

```
return {
            gateway: allowedPaymentMethods[0]?.tokenizationSpecification?.parameters?.gateway || this.getGatewayIdentifier(),
            gatewayMerchantId,
        };
```